File: .github/skills/skill-authoring/tools/validate-skill.py
#!/usr/bin/env python3
"""
Validate a skill directory for required files and structure.
"""
import os
import sys

if len(sys.argv) != 2:
    print("Usage: validate-skill.py <skill-dir>")
    sys.exit(1)

skill_dir = sys.argv[1]
required = ["SKILL.md", "onboarding.md", os.path.join("examples", "README.md")]
missing = [f for f in required if not os.path.exists(os.path.join(skill_dir, f))]

if missing:
    print(f"Missing files: {', '.join(missing)}")
    sys.exit(2)
print("Skill structure valid.")

File: .github/skills/skill-authoring/tools/validate-skill.mjs
// validate-skill.mjs
import fs from 'fs';
import path from 'path';

const skillDir = process.argv[2];
if (!skillDir) {
  console.error('Usage: node validate-skill.mjs <skill-dir>');
  process.exit(1);
}
const required = ['SKILL.md', 'onboarding.md', path.join('examples', 'README.md')];
const missing = required.filter(f => !fs.existsSync(path.join(skillDir, f)));
if (missing.length) {
  console.error('Missing files:', missing.join(', '));
  process.exit(2);
}
console.log('Skill structure valid.');
